zkevm-circuits/src/copy_table.rs

```diff
@@ -0,0 +1,620 @@
+#![allow(missing_docs)]
+use bus_mapping::circuit_input_builder::{CopyDataType, CopyEvent, CopyStep, NumberOrHash};
+use eth_types::{Field, ToAddress, ToScalar, U256};
+use gadgets::{
+    binary_number::{BinaryNumberChip, BinaryNumberConfig},
+    less_than::{LtChip, LtConfig, LtInstruction},
+    util::{and, not, or, Expr},
+};
+use halo2_proofs::{
+    circuit::{Layouter, Region},
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Fixed, Selector, VirtualCells},
+    poly::Rotation,
+};
+
+use crate::evm_circuit::{
+    table::{BytecodeFieldTag, LookupTable, RwTableTag, TxContextFieldTag, TxLogFieldTag},
+    util::{constraint_builder::BaseConstraintBuilder, RandomLinearCombination},
+    witness::Block,
+};
+
+/// The rw table shared between evm circuit and state circuit
+#[derive(Clone, Debug)]
+pub struct CopyTableConfig<F> {
```

Comment:
I was in two minds since the design doc used the words "table lookup" and "circuit" interchangeably. I do support the renaming to `copy_circuit.rs` and `CopyCircuit` 👍